src/app/core/directives/product-context.directive.ts

```diff
@@ -1,34 +1,70 @@
+import { Directive, Input, OnInit, Optional, Output, SkipSelf } from '@angular/core';
+import { ReplaySubject, combineLatest, distinctUntilChanged, pairwise, startWith } from 'rxjs';
+
 import {
-  Directive,
-  Input,
-  OnChanges,
-  OnDestroy,
-  OnInit,
-  Optional,
-  Output,
-  SimpleChanges,
-  SkipSelf,
-} from '@angular/core';
-
-import { ProductContextDisplayProperties, ProductContextFacade } from 'ish-core/facades/product-context.facade';
+  ProductContext,
+  ProductContextDisplayProperties,
+  ProductContextFacade,
+} from 'ish-core/facades/product-context.facade';
 import { ProductCompletenessLevel, SkuQuantityType } from 'ish-core/models/product/product.model';
 
+declare type IdType = number | string;
+
 @Directive({
   selector: '[ishProductContext]',
   providers: [ProductContextFacade],
+  exportAs: 'ishProductContext',
 })
-export class ProductContextDirective implements OnInit, OnChanges, OnDestroy {
+export class ProductContextDirective implements OnInit {
   @Input() completeness: 'List' | 'Detail' = 'List';
-  @Input() propagateIndex: number;
-
   @Output() skuChange = this.context.select('sku');
   @Output() quantityChange = this.context.select('quantity');
 
-  constructor(
-    @SkipSelf() @Optional() private parentContext: ProductContextFacade,
-    private context: ProductContextFacade
-  ) {
-    this.context.hold(this.context.$, () => this.propagate());
+  private propIndex = new ReplaySubject<IdType>(1);
+
+  constructor(@SkipSelf() @Optional() parentContext: ProductContextFacade, private context: ProductContextFacade) {
+    if (parentContext) {
+      function removeFromParent(parent: ProductContext['children'], id: IdType) {
+        delete parent[id];
+      }
+
+      function addToParent(parent: ProductContext['children'], id: IdType, context: ProductContext) {
+        parent[id] = context;
+      }
+
+      function isId(id: IdType): boolean {
+        return id !== undefined;
+      }
+
+      parentContext.connect(
+        'children',
+        combineLatest([
+          this.propIndex.pipe(startWith(undefined), distinctUntilChanged(), pairwise()),
+          this.context.select().pipe(distinctUntilChanged()),
```

Comment:
TODO: 
- `distinctUntilChanged` does not work for whole context object
- add `debounceTime` for stable lazy property upgrades

Reply:
done 👍